### DISCUSSION TITLE
Set value without option list reloading

### DISCUSSION THREAD

**Discussion by jelly-garden** [sa-si-dev/virtual-select#67] [Q&A] [answered]
When I call setValue(), option list reloading is excuted automatically.

I want to make group option selectable.
![image](https://user-images.githubusercontent.com/60183166/135388672-5670cfaa-8bd6-4e86-bf24-fbb0752f3edf.png)

I tried to realize that with two way.
1. create group option's  label renderer with checkbox.
2. add padding-left to member(child) option.

But there is a problem because of the option list reloading that happens automatically.

Is there any way to solve this?

**Comment by sa-si-dev** [comment, marked as answer]
@jelly-garden this is expected behavior only. The options list would be rerendered on value select and on the scrolling options list.

The purpose of this plug-in is, to improve the performance by reducing the no.of DOM elements at a time.
To achieve this, only a few options would be rendered at any point in time. And it would be rerendered when needed.

Regarding your requirement,  this plug-in already has a feature to group options ([example](https://sa-si-dev.github.io/virtual-select/#/examples?id=option-group)), but the only missing part is, there is no checkbox in the group title. I will implement it in the future release


<img width="272" alt="Screenshot 2021-10-01 at 6 53 29 PM" src="https://user-images.githubusercontent.com/76465932/135627325-5b247441-2460-4037-92af-36d8f2dc7c9b.png">

**Reply by sa-si-dev** [reply]
@jelly-garden, this feature has been implemented in the [release v1.0.17](https://github.com/sa-si-dev/virtual-select/releases/tag/v1.0.17)

Now checkbox would be shown with the group titles which could be used to select/deselect all child options.

**Reply by jelly-garden** [reply]
Thanks a lot. That was extremely helpful!
